src/features/antora/antoraConfigs.ts

```diff
@@ -0,0 +1,97 @@
+import vscode, { CancellationTokenSource, FileType } from 'vscode'
+import { fastFindFiles } from './fastFindFiles'
+import yaml from 'js-yaml'
+import ospath from 'path'
+
+export class AntoraConfig {
+  public contentSourceRootPath: string
+  public contentSourceRootFsPath: string
+
+  private static versionMap = new Map<string, number>()
+
+  constructor (public uri: vscode.Uri, public config: { [key: string]: any }) {
+    const path = uri.path
+    this.contentSourceRootPath = path.slice(0, path.lastIndexOf('/'))
+    this.contentSourceRootFsPath = ospath.dirname(uri.fsPath)
+    if (config.version === true || config.version === undefined) {
+      config.version = this.getVersionForPath(path)
+    }
+  }
+
+  public getVersionForPath (path: string): string {
+    const version = AntoraConfig.versionMap.get(path)
+    if (version) return `V-${version}`
+
+    const nextVersion = AntoraConfig.versionMap.size + 1
+    AntoraConfig.versionMap.set(path, nextVersion)
+    return `V-${nextVersion}`
+  }
+}
+
+const cache = {
+  antoraConfigs: [] as AntoraConfig[],
+  antoraConfigUris: [] as vscode.Uri[],
+}
+
+let ongoingRefreshPromise : Promise<void> | undefined
+
+async function refreshAntoraConfigs (token?: vscode.CancellationToken) {
+  if (!ongoingRefreshPromise) {
```

Comment:
Used a flag to prevent launching multiple refreshes in parallel: will return the existing refreshPromise if there is an ongoing cache refresh